packages/@ember/-internals/metal/lib/array.ts

```diff
@@ -17,11 +20,17 @@ export function objectAt<T>(array: T[] | EmberArray<T>, index: number): T | unde
 }
 
 export function replace<T>(
-  array: NativeArray<T> | MutableArray<T>,
+  array: T[] | MutableArray<T>,
```

Comment:
Since we're checking `Array.isArray` below we should obviously be allowing any array.